batch/batch/worker/worker.py

```diff
@@ -529,6 +530,49 @@ def __str__(self):
         return f'container {self.job.id}/{self.name}'
 
 
+class JVMProcess:
+    classpath = f'{find_spark_home()}/jars/*:/hail.jar'
+    stack_size = 512 * 1024
+    thread_pool = None
+
+    def __init__(self, job, main_spec):
+        self.job = job
+        self.heap_size = main_spec['memory'] - self.stack_size
+        self.command = ['java', '-classpath', self.classpath, f'-Xmx:{self.heap_size}', f'-Xss:{self.stack_size}']
+        self.command.extend(main_spec['command'])
+        self.env = {e['name']: e['value'] for e in main_spec['env']}
+
+        self.timing = {}
+
+    async def run(self):
+        log.info(f'running {self}')
+        self.timing['start_time'] = time_msecs()
+        # run jvm job here
+        await check_shell_output(' '.join(self.command))
```

Comment:
We can avoid the shell:
```python3
proc = await asyncio.create_subprocess_exec(
    *self.command,
    stdout=asyncio.subprocess.PIPE,
    stderr=asyncio.subprocess.PIPE)
out, err = await proc.communicate()
# ...
return proc.returncode, out, err
```

Reply:
actually we probably want to redirect err to out so we can upload the whole thing to GCS as one file:
```
stderr=asyncio.subprocess.STDOUT
```

Reply:
This also gives you a `proc` that you can `kill()`